### PR TITLE
Added cobra support and deprecated non-posix flags

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

It represents initial support for using cobra commands/subcommands versus plain flag usage for process-agent.

### Motivation

This is in preparation for adding subcommands to change runtime configuration and runtime debugging.

### Describe how to test your changes

process-agent should still start and support the same command line parameters as before:
All OSes: 
- -config
- -ddconfig
- -sysprobe-config
- -info
- -verison
- -check

MacOS/Linux Specific:
- -pid

Windows Specific:
- -install-service
- -uninstall-service
- -start-service
- -stop-service
- -foreground

Additionally, the shorthand commands should also work ("-i", "-v", and, on mac/linux: "-p")

The behavior between two dashes (i.e. --pid vs -pid) should be the same, except the non-posix version (the one with only one "-" character) should display a warning.

It is important that both windows and linux/macos is tested, since their initialization sequence is different.

### Checklist
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.

### CODE REVIEW

File: releasenotes/notes/process-agent-cobra-4be0f08037b8141e.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
```
Comment (xornivore) [inline releasenotes/notes/process-agent-cobra-4be0f08037b8141e.yaml:7]: nit - you can drop this header, it's just to help with filling out various sections.

**Review by xornivore — APPROVED** [review]
Looks great, small nit for the release note, but :shipit:

File: releasenotes/notes/process-agent-cobra-4be0f08037b8141e.yaml (hunk 1 above)
Comment (just-chillin) [reply releasenotes/notes/process-agent-cobra-4be0f08037b8141e.yaml:7]: Oh alright!

File: cmd/process-agent/main_common.go [hunk 2]
```diff
@@ -27,6 +22,11 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/profiling"
 	"github.com/DataDog/datadog-agent/pkg/version"
+	"github.com/spf13/cobra"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"time"
```
Comment (shang-wang) [inline cmd/process-agent/main_common.go:29]: nit: the import format seems to be messed up here.

File: cmd/process-agent/main_common.go [hunk 3]
```diff
@@ -50,6 +50,36 @@ var (
...
+
+		// Windows flags
+		"-install-service": {}, "-uninstall-service": {}, "-start-service": {}, "-stop-service": {}, "-foreground": {},
+	}
+
+	for i, a := range os.Args {
+		if _, ok := deprecatedFlags[a]; !ok {
+			continue
+		}
+		fmt.Printf("WARNING: `%s` argument is deprecated and will be removed in a future version. "+
```
Comment (shang-wang) [inline cmd/process-agent/main_common.go:77]: nit: let's not use `+` for the new line, you can put everything in one line.

File: cmd/process-agent/main_common.go [hunk 4]
```diff
@@ -50,6 +50,36 @@ var (
...
+	deprecatedFlags := map[string]struct{}{
+		// Global flags
+		"-config": {}, "-ddconfig": {}, "-sysprobe-config": {}, "-pid": {}, "-info": {}, "-version": {}, "-check": {},
+
+		// Windows flags
+		"-install-service": {}, "-uninstall-service": {}, "-start-service": {}, "-stop-service": {}, "-foreground": {},
+	}
+
+	for i, a := range os.Args {
+		if _, ok := deprecatedFlags[a]; !ok {
```
Comment (shang-wang) [inline cmd/process-agent/main_common.go:74]: nit: I believe you can use `if _, ok := deprecatedFlags[a]; ok` to remove the `continue` line.

Comment (just-chillin) [reply cmd/process-agent/main_common.go:74]: I think this is a stylistic thing for me. I kind of think about it as filtering out the values that you don't want. Anyway, I went ahead and fixed it.

File: cmd/process-agent/main_common.go (hunk 3 above)
Comment (just-chillin) [reply cmd/process-agent/main_common.go:77]: Fixed!

File: cmd/process-agent/main_common.go (hunk 2 above)
Comment (just-chillin) [reply cmd/process-agent/main_common.go:29]: Fixed!

File: cmd/process-agent/main_common.go [hunk 5]
```diff
@@ -50,6 +50,36 @@ var (
...
+	},
+}
+
+// fixDeprecatedFlags modifies os.Args so that non-posix flags are converted to posix flags
+// it also displays a warning when a non-posix flag is found
+func fixDeprecatedFlags() {
+	deprecatedFlags := map[string]struct{}{
+		// Global flags
+		"-config": {}, "-ddconfig": {}, "-sysprobe-config": {}, "-pid": {}, "-info": {}, "-version": {}, "-check": {},
+
```
[the remark this answers is not in the document]
Comment (just-chillin) [reply cmd/process-agent/main_common.go:68]: Fixed

**Review by shang-wang — APPROVED** [review]

File: cmd/process-agent/main.go [hunk 6]
```diff
@@ -3,29 +3,27 @@
...
 
-	flag.StringVar(&opts.pidfilePath, "pid", "", "Path to set pidfile for process")
-	flag.BoolVar(&opts.info, "info", false, "Show info about running process agent and exit")
-	flag.BoolVar(&opts.version, "version", false, "Print the version and exit")
-	flag.StringVar(&opts.check, "check", "", "Run a specific check and print the results. Choose from: process, connections, realtime")
-	flag.Parse()
+	rootCmd.PersistentFlags().StringVarP(&opts.pidfilePath, "pid", "p", "", "Path to set pidfile for process")
+	rootCmd.PersistentFlags().BoolVarP(&opts.info, "info", "i", false, "Show info about running process agent and exit")
+	rootCmd.PersistentFlags().BoolVarP(&opts.version, "version", "v", false, "Print the version and exit")
+	rootCmd.PersistentFlags().StringVar(&opts.check, "check", "", "Run a specific check and print the results. Choose from: process, connections, realtime")
```
Comment (shang-wang) [inline cmd/process-agent/main.go:23]: This command explanation is actually not correct, the correct list should be
```
process rtprocess container rtcontainer connections pod
```

Comment (just-chillin) [reply cmd/process-agent/main.go:23]: Should this be added into a seperate pr?

Comment (just-chillin) [reply cmd/process-agent/main.go:23]: Curious though, since the original description is the same. Why was this wrong in the first place?

Comment (xornivore) [reply cmd/process-agent/main.go:23]: Probably just never was updated, so a separate PR makes sense, thank you.

**Pull request merged**